eval/src/vespa/eval/instruction/join_with_number_function.cpp

```diff
@@ -98,14 +119,12 @@ JoinWithNumberFunction::optimize(const TensorFunction &expr, Stash &stash)
             const ValueType &result_type = join->result_type();
             const TensorFunction &lhs = join->lhs();
             const TensorFunction &rhs = join->rhs();
-            if (lhs.result_type().is_double() &&
-                (result_type == rhs.result_type()))
-            {
+            if (lhs.result_type().is_double()) {
+                assert(result_type.dimensions() == rhs.result_type().dimensions());
```

Comment:
these asserts are just weaker forms of the input_type.map() -> output_type assert in compile_self. consider removing them (or use the same assert for consistency).